src/frontend/js/pages/DashboardCourses/useOrdersEnrollments.tsx

```diff
@@ -0,0 +1,277 @@
+ *
+ * The algorithm of integrity building is:
+ * - Fetch page P of orders and enrollments, merge them, and sort them. We call it `previousStack`
+ * - Fetch page P+1 of orders and enrollments, merge them, and sort them. We call it `currentStack`
+ * - If `previousStack` and `currentStack` are equal on [0, wantedIntegrityCount],
+ *    we can say that the integrity is verified over [0, wantedIntegrityCount], because it means this
+ *    portion will never change over futures fetches ( due to consistent sorting ).
+ * - If not, then loop over the next pages until we find the integrity.
+ * - Then provide the slice [0, cursor] of the stack with integrity via data.
+ * - NB: If previousStack and currentStack have the same length, it means that we fetched all
+ *    entities, so we can say that the integrity is verified over [0, totalCount] by definition.
+ *
+ * Good to know:
+ * - The first time useOrdersEnrollments is called, it will fetch the first page of orders and
+ *    enrollments automatically.
+ * - Then, subsequent calls to `next` will be needed to get the next slices of data.
+ */
+export const useOrdersEnrollments = () => {
+  const { fetchEntitiesInStack } = useUtils();
+  const [error, setError] = useState<Maybe<string>>();
+  const [cursor, setCursor] = useState(CHUNK_SIZE);
+  const [integrityCount, setIntegrityCount] = useState(0);
+  const [stack, setStack] = useState<Data[]>([]);
+  const [page, setPage] = useState(0);
+  const [totalCount, setTotalCount] = useState<number | undefined>();
+  const [isSyncing, setIsSyncing] = useState(false);
+  const intl = useIntl();
+
+  const syncIntegrityCount = async (wantedIntegrityCount: number) => {
+    if (integrityCount === wantedIntegrityCount) {
+      return;
+    }
+    setIsSyncing(true);
+    log('We want integrity', wantedIntegrityCount, 'page', page);
+
+    let pageToLoad = page;
+    let previousStack = [...stack];
+    let currentStack: Data[];
+    let newTotalCount;
+    while (true) {
+      pageToLoad += 1;
+      currentStack = [...previousStack];
+      log('Fetching page ' + pageToLoad + ' ...');
+
+      try {
+        // eslint-disable-next-line no-await-in-loop
+        const { totalCount: tmpTotalCount } = await fetchEntitiesInStack(pageToLoad, currentStack);
+        if (typeof newTotalCount === 'undefined') {
+          newTotalCount = tmpTotalCount;
+        }
+        log(
+          'checkIntegrity',
+          'previous',
+          previousStack.length,
+          'current',
+          currentStack.length,
+          'newTotalCount',
+          newTotalCount,
+        );
+
+        // Having the same length means that we reached the end of the list.
+        if (currentStack.length === previousStack.length) {
+          setIntegrityCount(currentStack.length);
+          break;
+        }
+        if (checkIntegrity(previousStack, currentStack, wantedIntegrityCount)) {
+          setIntegrityCount(wantedIntegrityCount);
+          break;
+        }
```

Comment:
Nope! `currentStack` also includes all the entities fetched from the start. ( To guarantee integrity we merge then re-sort everything in case where a newly fetched entity would be thrown in front of the array )